osu.Game/IPC/IpcServer.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Platform;
+
+#nullable enable
+
+namespace osu.Game.IPC
+{
+    /// <summary>
+    /// Provides IPC between osu! instances.
+    /// </summary>
+    [Cached]
+    public class IpcServer : CompositeDrawable
+    {
+        private readonly IIpcHost host;
+        private readonly Dictionary<string, ILoadableIpcChannel> channels = new Dictionary<string, ILoadableIpcChannel>();
```

Comment:
as the code quality checks helpfully point out, this is never populated and channels are loaded every time, even if the type is the same

Reply:
Fixed. Odd that I didn't notice it while testing using draw visualiser.